RockLib.Configuration.ObjectFactory/ConfigReloadingProxy.cs

```diff
@@ -92,14 +93,28 @@ protected ConfigReloadingProxy(IConfiguration section, DefaultTypes defaultTypes
       /// </summary>
       public event EventHandler? Reloaded;
 
+      /// <summary>
+      /// Standard implementation of Dispose
+      /// </summary>
+      public void Dispose()
+      {
+          Dispose(true);
+          GC.SuppressFinalize(this);
+      }
+
       /// <summary>
       /// Dispose the underlying object if it implements <see cref="IDisposable"/>.
       /// </summary>
-#pragma warning disable CA1063 // Implement IDisposable Correctly
-#pragma warning disable CA1816 // Dispose methods should call SuppressFinalize
-      public void Dispose() => (Object as IDisposable)?.Dispose();
-#pragma warning restore CA1816 // Dispose methods should call SuppressFinalize
-#pragma warning restore CA1063 // Implement IDisposable Correctly
+      /// <param name="disposing"></param>
+      protected virtual void Dispose(bool disposing)
+      {
```

Comment:
I don't think it matters too much, but this technically doesn't implement the dispose pattern correctly.

You may want to add a `_disposed` variable and do the whole `if (_disposed) return;` stuff.

Reply:
For the unit test, it doesn't matter if it implements it correctly. Also, if the type is `sealed`, the .NET analyzers don't bark about the type with it implementing `IDisposable`.

Reply:
I mean the problem that can happen from disposing of an object twice ultimately is that the dispose can fail after the first time (like deleting a file or something). Chances are that the reload token has it's own multi-dispose protection, and that consumers of this will either have their own multi-dispose protection, or their downstream dependencies do.

The consumer will be the one in control of this object, so a consumer could forcefully call dispose on this more than once.

I'm just thinking better safe than sorry, and it's not much to change, but ultimately, i think the chances of it actually becoming an issue are very very very small.

Reply:
There are comments above this type from a previous comment where we didn't want to potentially "break" too much. That said, I just noticed the analyzers are saying these pragmas are no longer necessary, so....I'm going to remove them.